channels/auth.py

```diff
@@ -53,7 +55,7 @@ def get_user(scope):
 
 
 @database_sync_to_async
-def login(scope, user, backend=None):
+def login(scope: Dict[str, Any], user, backend=None) -> None:
```

Comment:
I think the type hinting also have another NoReturn type

Reply:
Good to know, I will work on that too

Reply:
What about Awaitable[None], it can be translated to Awaitable[NoReturn]?

Reply:
I discovered that NoReturn has been introduced from 3.5.4 and 3.6.2 so I think I will replace the NoReturn with None

Reply:
Do the `def ...(..) -> ` methods that are decorated with `database_sync_to_async` are not returning an `Awaitable`

the `@database_sync_to_async` is converting them to be `Awaitable`. So the return type of the above should be `-> None`